strapi/strapi
# feat: show hidden fields in history frontend

### What does it do?

Makes hidden fields visible in the History page. This is not about internal fields like id, but about the ones the user may have deleted from the layout via the configure the view page. I called them "remaining fields" in the code, to match what was already done in the code for that page.

It applies the fix in two places:

- At the root of the history page layout. This one was the most straight forward.
- Inside the components. We didn't have direct access to the component's rendering logic, so I had to refactor it to pass a new `renderLayout`. It has the benefit of avoiding duplicating logic between repeatable and non-repeatable components, and avoids some prop drilling for `renderInput`.

I want to write e2e tests for this, they have issues we need to fix first.

### Why is it needed?

Because even though they're not visible in the edit view, the fields that aren't in the layout would still be restored if the history version is restored. It's therefore important to show them regardless, so that users know what they're restoring.

### How to test it?

Go to the configure the view page for a content type. Delete a root field there (not in the CTB), as well as a field that's inside a component.

Now open the history page for an entry of that content type. You should see the deleted root field on a new line at the bottom of the page, but above the unknown fields.

Also check that component inputs work as expected in the Edit View page, since their code was refactored.
